app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+    if params[:pet_name].present?
+      @pets = Pet.search(params[:pet_name])
+    else
+      @pets = @application.pet_search(params[:pet_name])
```

Comment:
I'm curious what led you to this solution with `pet_search`. It seems it will only be reached with the `params[:pet_name]` is not present, which also means there are no search terms available yet. So, in the view, you can display the seach results only if @pets is available. I don't quite understand why you are populating @pets in this case when the search term isn't present